binary-search.py

```diff
@@ -1503,7 +1503,7 @@ def main():
              rate = 100.0
         else:
              rate = 9999 / ((t_global.args.frame_size) * 8 + 64 + 96.0)
-    initial_rate = rate
+    initial_rate = True
```

Comment:
Is there a reason why the behavior of initial_rate is being changed?  This seems completely separate from how the next rate is being calculated.

Reply:
The variable initial_rate is only used to check if we passed in the first trial (therefore, we double the rate and retry), but we do it in a very convoluted way:

1) Initialization:
    line 1506: intial_rate = rate
    line 1509: prev_fail_rate = rate

2) First pass in the big while loop (final_validation is False):
    line 2027: next_rate = (prev_fail_rate + rate) / 2      # since prev_fail_rate == rate, next_rate = rate
    line 2029: final_validation = True
    line 2036: rate = nex_rate    # that is, rate remains unchanged

3) Second pass in the big while loop:
    line 2000: if initial_rate == rate: bs_logger("Detected requested rate is too low, doubling rate and restarting search")

The reason why I changed it here is because I'm using tx_rate instead of rate on line 2027, therefore rate WILL NOT remain unchanged between the 1st and 2nd passes and we end up failing to detect that the rate is too low on line 2000.
Also, using a boolean to track that we passed in the first trial seems more clear. I guess we might also rename initial_rate to initial_pass?